hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2972,7 +3011,7 @@ func (r *HostedClusterReconciler) reconcileMachineApprover(ctx context.Context,
 		}
 		deployment := machineapprover.Deployment(controlPlaneNamespaceName)
 		if _, err := createOrUpdate(ctx, r.Client, deployment, func() error {
-			return reconcileMachineApproverDeployment(deployment, hcluster, sa, kubeconfigSecretName, config, image, r.AvailabilityProberImage)
+			return reconcileMachineApproverDeployment(deployment, hcluster, sa, kubeconfigSecretName, config, image, r.AvailabilityProberImage, r.SetSecurityContext)
```

Comment:
why do we use 3 different terms to express the same: SetSecurityContext, SetSecurityContextNonRoot, explicitNonRootSecurityContext? Can we please align?

Reply:
I agree, we should align here. Updated to use `SetDefaultSecurityContext` instead. https://github.com/openshift/hypershift/pull/846/commits/17d7cd45a2a197b4478808bf2d1cfbd0fdb9a8a2